streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1086,6 +1089,34 @@ private boolean nodeGroupContainsGlobalSourceNode(final Set<String> allNodesOfGr
         return false;
     }
 
+    private static class NodeComparator implements Comparator<TopologyDescription.Node>, Serializable {
+
+        @Override
+        public int compare(final TopologyDescription.Node node1,
+                           final TopologyDescription.Node node2) {
+            final int size1 = ((AbstractNode) node1).size;
+            final int size2 = ((AbstractNode) node2).size;
+
+            // it is possible that two nodes have the same sub-tree size (think two nodes connected via state stores)
+            // in this case default to processor name string
+            if (size1 != size2) {
+                return size2 - size1;
```

Comment:
From String#compareTo()
```
        while (k < lim) {
            char c1 = v1[k];
            char c2 = v2[k];
            if (c1 != c2) {
                return c1 - c2;
```
Should the size comparison follow the same ordering ?
i.e. size1 - size2

Reply:
Actually no, because for `String` comparison we prefer the strings with "higher" characters, while for NodeComparator we prefer nodes with "smaller" size in order to print them first.